pytorch_lightning/accelerator_backends/gpu_backend.py

```diff
@@ -28,11 +28,14 @@ class GPUBackend(object):
     def __init__(self, trainer):
         self.trainer = trainer
 
-    def setup(self, model):
+    def setup(self, model, datamodule=None):
```

Comment:
no need to pass these in...

in fit:
self.datamodule = datamodule.

Then in these methods check trainer.datamodule